### PR TITLE
Feature/derived quantities

### PR DESCRIPTION
**Pull request by ryan-kipawa**

### CODE REVIEW

**Comment by ryan-kipawa** [comment]
The basic structure and functionality is complete:

- Derived quantities work with nearly the same API as if they were real quantities
- A set of default derived quantities is provided (complete for collection systems, WD needs additions)
- Possibility for users to extend with their own derived quantities.

I suggest any further work (i.e. additional default derived quantities) be in another PR.

**Comment by ryan-kipawa** [comment]
@gedaskir I removed the DerivedQuantityManager to simplify it a bit. Now derived quantities ultimately live on Res1D as a list, and are propagated through the network with add/remove methods

File: tests/testdata/NetworkFactorTwo.res1d
Comment (gedaskir) [inline tests/testdata/NetworkFactorTwo.res1d:1]: Does this file need to be source controlled? I think it is being added as part of a notebook run. Likely gitignore has it missing.

File: tests/testdata/NetworkNodeWaterLevel.dfs0
Comment (gedaskir) [inline tests/testdata/NetworkNodeWaterLevel.dfs0:1]: Does this file need to be source controlled?

**Pull request closed without merging**